crates/node-builder/src/builder.rs

```diff
@@ -157,11 +156,11 @@ type RethFullAdapter<DB, N> =
 /// [builder]: https://doc.rust-lang.org/1.0.0/style/ownership/builders.html
 pub struct NodeBuilder<DB, State> {
     /// All settings for how the node should be configured.
-    config: NodeConfig,
+    pub config: NodeConfig,
     /// State of the node builder process.
-    state: State,
+    pub state: State,
     /// The configured database for the node.
-    database: DB,
+    pub database: DB,
```

Comment:
if we make these pub then the launch trait is kinda pointless?

Reply:
I don't think so because devs need access to these fields within the launch trait. For example, the second line of code in the default launch trait impl deconstructs `NodeBuilder` then uses the database to create the node's `ProviderFactory`.

If we make these fieldss private, then that forces devs to impl their own node builder struct and they lose access to a lot of great methods/features `NodeBuilder` provides

Reply:
@mattsse I restored the fields to private and added `WithLaunchContext::launch_with`